### DISCUSSION TITLE
Inverter - Wake up from idle

### DISCUSSION THREAD

**Discussion by stringkillersa** [wills106/homeassistant-solax-modbus#49] [Q&A]
Has anyone found a way to wake up the inverter when runmode is idle?
So far the only working way i found is NOT allowing it to fall asleap.

**Comment by wills106** [comment]
There is the battery_awaken button. But I can't verify if it works or not on the Gen3. It apparently works on the Gen2 though?

I have enough battery to go just over a full 24hrs so mine nether goes idle / to sleep.

**Reply by stringkillersa** [reply]
@wills106  Mines are G4. Yes, that was the first obvious test - Battery awaken, but unfortunately - it did'nt work...

**Reply by mickemartinsson** [reply]
@wills106 Mine is a G3 and it doesn't work. We got snow again... and with an electric car that uses up the battery faster, mine seems to go idle and perhaps that is the problem with Issue #48??

**Reply by stringkillersa** [reply]
@mickemartinsson In my case (Similar to #48)  was due to idle mode.

**Reply by wills106** [reply]
I think your issue #48 is possibly related to the Inverter going idle.

I explain in [this issue here](https://github.com/wills106/homsassistant-solax-modbus/issues/3#issuecomment-1022044481) that I think some of the behaviours experience seem to be Inherent from normal PV only Inverters.

I think the battery_awaken is only relevant to the Gen2 I am pretty sure I tried it before adding extra batteries? But it was a while go.

There must be something missing to wake the Inverter up form Modbus as I am sure you can change the charger_use_mode from the cloud / Inverter it's self but it takes 15 - 30s for it to respond / wake up

If I know I will run out battery I will put the Inverter into Force Time Use before it runs out.
I am on an electric Tariff where there is a cheap period during the night from 03:30 - 06:30 so I leave a time period set. If I need more battery capacity I would alter the times.
![image](https://user-images.githubusercontent.com/18155231/161702351-a7d72a1f-c94b-4ba2-9ec1-e34f3a777657.png)
Then normally keep it in Self Use Mode and switch to Force Time Use when needed.

**Reply by hwvvse** [reply]
with g3 , if in idle mode, system switch on and off, then you have a min or so to change, just on the inverter tho...

**Reply by stringkillersa** [reply]
@hwvvse Yeah, I noticed this.  Yesterday I changed to use Backup_mode instead of Self_use as these modes are quite similar, setting a lower SOC for backup mode to 20%. This kept the inverter awake for the first time and it was in runmode "Standby" instead of Idle. So my automations during night worked as they should, changing modes and so forth

**Reply by stringkillersa** [reply]
I'm adding a comment here for others who have problems with their inverter going to sleep. Having the Inverter set in Selfuse-mode will cause the inverter to enter sleepmode some time after the solar productions turn zero - ie, during night or snow covering the panels. My solution to this is to use the "backup mode" instead. It is very similar to Selfuse, but the inverter is always awake to be able to supply backup.  My automation then uses charge/discharge current to select wether it actually SHOULD charge or discharge,

**Comment by mickemartinsson** [comment]
When I’m using Backup Mode the inverter always buys a small amount to charge the battery. When I use Self Use Mode it only buys to even the phases, not to charge/maintain battery SOC. I’m switching between Self Use Mode and Forced Mode, no problem with night sleep on inverter but did have some problems with snow last winter and that it went to sleep.

Från: Stig Andersson ***@***.***>
Datum: fredag, 19 augusti 2022 11:08
Till: wills106/homeassistant-solax-modbus ***@***.***>
Kopia: Mikael Martinsson ***@***.***>, Mention ***@***.***>
Ämne: Re: [wills106/homeassistant-solax-modbus] Inverter - Wake up from idle (Discussion #49)

I'm adding a comment here for others who have problems with their inverter going to sleep. Having the Inverter set in Selfuse-mode will cause the inverter to enter sleepmode some time after the solar productions turn zero - ie, during night or snow covering the panels. My solution to this is to use the "backup mode" instead. It is very similar to Selfuse, but the inverter is always awake to be able to supply backup. My automation then uses charge/discharge current to select wether it actually SHOULD charge or discharge,
